### ISSUE TITLE
Getting "Error: Assertion failed" when trying to sign transaction

### ISSUE THREAD

**Issue by n1rna** [moneybutton/bsv#160]
Hi,

I found this issue and I can't seem to mitigate it in any way. I have tried different things that I will explain below.

The stack trace for the error looks like:

```
Uncaught (in promise) Error: Assertion failed
    at n (bn.js:5)
    at o.toBuffer (bn.js:485)
    at e.s.toBuffer.s.toDER (signature.js:158)
    at h.addSignature (publickeyhash.js:71)
    at x.applySignature (transaction.js:1233)
    at transaction.js:1198
    at Array.forEach (<anonymous>)
    at Object.each (_.js:62)
    at x.sign (transaction.js:1197)
    at Fe (transaction.js:152)
```

Where `transaction.js:152` looks something like:

```
tx.addOutput(BSV.Transaction.Output({ script, satoshis: 0 }))
    .change(change)
    .feePerKb(feePerKb);
    .sign(pvKey)
```

The problem does not occur when I test it on my local machine, but when I build the react app and deploy it on production, I keep getting this error which seems to be related to the `Buffer` type's presence. I have tried adding these lines to my react app's entry script (`index.js`):

```
import { Buffer } from 'buffer';

global.Buffer = Buffer; 
window.Buffer = Buffer;
```

But it didn't solve the problem. I don't know if I am doing something wrong or it's the `bsv` library, but I tried to upgrade the `bsv` package version to the latest version and it didn't solve the problem either.

So, any insights or helps are appreciated. Thanks.

**Comment by kevinejohn** [comment]
I ran into this issue and it seemed to be fixed after reloading node_modules

```
rm -rf node_modules
npm install
```

**Comment by conorbros** [comment]
Was there a solution to this? I tried removing node_modules and it didn't work

**Comment by 032Amit** [comment]
Hi @n1rna ,
Did you get any solution for this issue, i am also getting same issue. I tried to resolve it by updating bitcore-lib but no luck.

Thanks in advance.

**Comment by Linux-Server** [comment]
> Was there a solution to this? I tried removing node_modules and it didn't work

did u got the answer?

**Comment by conorbros** [comment]
> > Was there a solution to this? I tried removing node_modules and it didn't work
> 
> did u got the answer?

I had to serve the development build to get it to work. I'm guessing it was an issue with whatever the react build scripts were doing to bundle everything in the production build.

**Comment by ty-everett** [comment]
I solved this by using `react-scripts@^3.4.1` instead of `react-scripts@^4.0.0` in my `package.json`.

You will need to delete `package-lock.json` and `node_modules`, then run `npm i`.

You lose hot module reloading, and some other nice things, but it didn't bother me at all in my case. I always thought HMR was still buggy.

The reason this worked is that Webpack 5 removes a lot of default things, including `Buffer`, which are needed by BN.js. React scripts 3.x still uses Webpack 4.

**Comment by SubhajitSK** [comment]
To resolve the issue, you can try updating the **bn.js** package by running the command `npm install bn.js` in your project directory. After updating, make sure to bundle the file again to incorporate the changes.
